### PR TITLE
Handle multiple user-agent

### PR DESCRIPTION
**Pull request by nicoja**
Related to issue : https://github.com/spatie/robots-txt/issues/28

### CODE REVIEW

File: tests/RobotsTxtTest.php [hunk 1]
```diff
@@ -128,4 +128,60 @@ public function the_disallows_user_agent_check_is_case_insensitive()
         $this->assertFalse($robots->allows('/no-agents', 'UserAgent007'));
         $this->assertFalse($robots->allows('/no-agents', strtolower('UserAgent007')));
     }
+
+    /** @test */
+    public function it_can_handle_multiple_user_agent()
```
Comment (freekmurze) [inline tests/RobotsTxtTest.php:133]: Can this test be made smaller? Due to the sheer length, it's hard to see what is being tested. Can you test the same behaviour with less user agents?

File: src/RobotsTxt.php [hunk 2]
```diff
@@ -126,36 +126,49 @@ protected function getDisallowsPerUserAgent(string $content): array
...
-                $currentUserAgent = &$disallowsPerUserAgent[$this->parseUserAgent($line)];
+                $currentUserAgents[] = &$disallowsPerUserAgent[$this->parseUserAgent($line)];
 
                 continue;
             }
 
-            if ($currentUserAgent === null) {
+            if ($this->isDisallowLine($line)) {
+                $treatAllowDisallowLine = true;
+            } elseif ($this->isAllowLine($line)) {
```
Comment (freekmurze) [inline src/RobotsTxt.php:152]: In general I don't like `else` statements. Can this be refactored?

File: src/RobotsTxt.php [hunk 3]
```diff
@@ -126,36 +126,49 @@ protected function getDisallowsPerUserAgent(string $content): array
...
         }
 
         return $disallowsPerUserAgent;
     }
 
-    protected function isCommentLine(string $line): bool
+    protected function isCommentOrEmptyLine(string $line): bool
     {
-        return strpos(trim($line), '#') === 0;
+        return strpos(trim($line), '#') === 0 || trim($line) === '';
```
Comment (freekmurze) [inline src/RobotsTxt.php:171]: I don't like complex if statements. Could you you split the if and use early returns?

**Review by freekmurze — CHANGES_REQUESTED** [review]

File: tests/RobotsTxtTest.php (hunk 1 above)
Comment (nicoja) [reply tests/RobotsTxtTest.php]: I split tests and remove some user-agents

File: src/RobotsTxt.php (hunk 2 above)
Comment (nicoja) [reply src/RobotsTxt.php]: refator done

File: src/RobotsTxt.php (hunk 3 above)
Comment (nicoja) [reply src/RobotsTxt.php]: split done

File: src/RobotsTxt.php [hunk 4]
```diff
@@ -173,6 +195,16 @@ protected function parseDisallow(string $line): string
         return trim(substr_replace(strtolower(trim($line)), '', 0, 8), ': ');
     }
 
+    protected function isDisallowLine(string $line): string
+    {
+        return trim(substr(str_replace(' ', '', strtolower(trim($line))), 0, 6), ': ') === 'disall';
```
Comment (freekmurze) [inline src/RobotsTxt.php:200]: Why not check the full word here?

**Review by freekmurze — COMMENTED** [review]
Almost there...

File: src/RobotsTxt.php (hunk 4 above)
Comment (nicoja) [reply src/RobotsTxt.php]: You right, modification pushed

Comment (nicoja) [reply src/RobotsTxt.php]: is it ok for you ?

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**